en/tutorials/rag-blueprint.md

```diff
@@ -0,0 +1,1475 @@
+These are relatively cheap to calculate, and will likely provide good enough ranking signals for the first-phase ranking.
+
+Running the command below will save a .csv-file with the collected features, which can be used to train a ranking model for the first-phase ranking.
+
+<pre>
+python eval/collect_pyvespa.py --collect_matchfeatures 
+</pre>
+
+Our output file looks like this:
+
+<style>
+/* keep the 1-pixel borders */
+table, th, td {
+  border: 1px solid black;
+  border-collapse: collapse;   /* so adjacent borders overlap cleanly */
+}
+
+/* width stays the same */
+th { width: 120px; }
+
+/* add breathing room for header and body cells */
+th, td {
+  padding: 6px 10px;   /* top-bottom 6 px, left-right 10 px — tweak as you like */
+}
+</style>
+<table>
+  <thead>
+    <tr>
+      <th>query_id</th>
+      <th>doc_id</th>
+      <th>relevance_label</th>
+      <th>relevance_score</th>
+      <th>match_avg_top_3_chunk_sim_scores</th>
+      <th>match_avg_top_3_chunk_text_scores</th>
+      <th>match_bm25(chunks)</th>
+      <th>match_bm25(title)</th>
+      <th>match_max_chunk_sim_scores</th>
+      <th>match_max_chunk_text_scores</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>alex_q_01</td>
+      <td>50</td>
+      <td>1</td>
+      <td>0.660597</td>
+      <td>0.248329</td>
+      <td>8.444725</td>
+      <td>7.717984</td>
+      <td>0.</td>
+      <td>0.268457</td>
+      <td>8.444725</td>
+    </tr>
+    <tr>
+      <td>alex_q_01</td>
+      <td>82</td>
+      <td>1</td>
+      <td>0.649638</td>
+      <td>0.225300</td>
+      <td>12.327676</td>
+      <td>18.611592</td>
+      <td>2.453409</td>
+      <td>0.258905</td>
+      <td>15.644889</td>
+    </tr>
+    <tr>
+      <td>alex_q_01</td>
+      <td>1</td>
+      <td>1</td>
+      <td>0.245849</td>
+      <td>0.358027</td>
+      <td>15.100841</td>
+      <td>23.010389</td>
+      <td>4.333828</td>
+      <td>0.391143</td>
+      <td>20.582403</td>
+    </tr>
+    <tr>
+      <td>alex_q_01</td>
+      <td>28</td>
+      <td>0</td>
+      <td>0.988250</td>
+      <td>0.278074</td>
+      <td>0.179929</td>
+      <td>0.197420</td>
+      <td>0.</td>
+      <td>0.278074</td>
+      <td>0.179929</td>
+    </tr>
+    <tr>
+      <td>alex_q_01</td>
+      <td>23</td>
+      <td>0</td>
+      <td>0.968268</td>
+      <td>0.203709</td>
+      <td>0.182603</td>
+      <td>0.196956</td>
+      <td>0.</td>
+      <td>0.203709</td>
+      <td>0.182603</td>
+    </tr>
+  </tbody>
+</table>
+
+Note that the `relevance_score` in this table is just the random expression we used in the `second-phase` of the `collect-training-data` rank profile, and will be dropped before training the model.
+
+### Training a first-phase ranking model
+
+As you recall, a first-phase ranking expression must be cheap to evaluate.
+This most often means a heuristic handwritten combination of match features, or a linear model trained on match features.
```

Comment:
I would add some more clarifications here, to make this more newbie friendly. Something in the lines of:
* above, we simply added those dimensions
* we'll want to weigh them differently, by importance
* in practice, these weights are taken from... inspiration :)
* we can be more scientific and use a model to predict the best combinations
* and now here's how you plug the model-detected weights into Vespa